src/shared/ask-password-api.c

```diff
@@ -139,7 +142,7 @@ static int add_to_keyring(const char *keyname, AskPasswordFlags flags, char **pa
         if (r < 0)
                 return r;
 
-        serial = add_key("user", keyname, p, n, KEY_SPEC_USER_KEYRING);
+        serial = add_key("user", keyname, p, n, KEY_SPEC_SESSION_KEYRING);
```

Comment:
That's the whole point about this commit (and issue #5522): Adding a key to user keyring results in a key that can not be access any more.